common/cpp/build/tests/Makefile

```diff
@@ -21,46 +20,46 @@
 # bunch of make variables.
 #
 
+include ../../../cpp_unit_test_runner/include.mk
 
-ADDITIONAL_TEST_LIBS_PREFIX := \
-	google_smart_card_common \
-
-ADDITIONAL_TEST_LIBS_SUFFIX := \
-	nacl_io \
-
-
-ADDITIONAL_TEST_DEPS = \
-	nacl_io \
-	google_smart_card_common:.. \
+ROOT_SOURCES_PATH := ../../src
+ROOT_SOURCES_SUBDIR := google_smart_card_common
+SOURCES_PATH := $(ROOT_SOURCES_PATH)/$(ROOT_SOURCES_SUBDIR)
 
-
-TEST_SOURCES_PATH := ../../src
-
-TEST_SOURCES_SUBDIR := google_smart_card_common
-
-TEST_SOURCES := \
-	$(TEST_SOURCES_PATH)/$(TEST_SOURCES_SUBDIR)/formatting_unittest.cc \
-	$(TEST_SOURCES_PATH)/$(TEST_SOURCES_SUBDIR)/logging/hex_dumping_unittest.cc \
-	$(TEST_SOURCES_PATH)/$(TEST_SOURCES_SUBDIR)/messaging/typed_message_router_unittest.cc \
-	$(TEST_SOURCES_PATH)/$(TEST_SOURCES_SUBDIR)/numeric_conversions_unittest.cc \
-	$(TEST_SOURCES_PATH)/$(TEST_SOURCES_SUBDIR)/requesting/async_request_unittest.cc \
-	$(TEST_SOURCES_PATH)/$(TEST_SOURCES_SUBDIR)/requesting/async_requests_storage_unittest.cc \
-	$(TEST_SOURCES_PATH)/$(TEST_SOURCES_SUBDIR)/value_conversion_unittest.cc \
-	$(TEST_SOURCES_PATH)/$(TEST_SOURCES_SUBDIR)/value_debug_dumping_unittest.cc \
-	$(TEST_SOURCES_PATH)/$(TEST_SOURCES_SUBDIR)/value_unittest.cc \
+SOURCES := \
+	$(SOURCES_PATH)/formatting_unittest.cc \
+	$(SOURCES_PATH)/logging/hex_dumping_unittest.cc \
+	$(SOURCES_PATH)/messaging/typed_message_router_unittest.cc \
+	$(SOURCES_PATH)/numeric_conversions_unittest.cc \
+	$(SOURCES_PATH)/requesting/async_request_unittest.cc \
+	$(SOURCES_PATH)/requesting/async_requests_storage_unittest.cc \
+	$(SOURCES_PATH)/value_conversion_unittest.cc \
+	$(SOURCES_PATH)/value_debug_dumping_unittest.cc \
+	$(SOURCES_PATH)/value_unittest.cc \
 
 ifeq ($(TOOLCHAIN),pnacl)
 
-TEST_SOURCES += \
-	$(TEST_SOURCES_PATH)/$(TEST_SOURCES_SUBDIR)/pp_var_utils/construction_unittest.cc \
-	$(TEST_SOURCES_PATH)/$(TEST_SOURCES_SUBDIR)/pp_var_utils/enum_converter_unittest.cc \
-	$(TEST_SOURCES_PATH)/$(TEST_SOURCES_SUBDIR)/pp_var_utils/struct_converter_unittest.cc \
-	$(TEST_SOURCES_PATH)/$(TEST_SOURCES_SUBDIR)/value_nacl_pp_var_conversion_unittest.cc \
+SOURCES += \
+	$(SOURCES_PATH)/pp_var_utils/construction_unittest.cc \
+	$(SOURCES_PATH)/pp_var_utils/enum_converter_unittest.cc \
+	$(SOURCES_PATH)/pp_var_utils/struct_converter_unittest.cc \
+	$(SOURCES_PATH)/value_nacl_pp_var_conversion_unittest.cc \
 
 endif
 
-ADDITIONAL_TEST_CPPFLAGS := \
-	-I$(TEST_SOURCES_PATH) \
+CXXFLAGS := \
+	-I$(ROOT_SOURCES_PATH)/ \
+	-pedantic \
+	-Wall \
+	-Werror \
+	-Wextra \
+	-std=gnu++11 \
```

Comment:
I would prefer `-std=c++11` (if that does not work, perhaps document the reason for it - not in this file, but in a more central place). Either thing should probably be a separate CL.

Reply:
In NaCl there were some problems with `c++11`, which is why is had to use `gnu++11` (don't remember specific examples right now).
But you're right that it's not documented anywhere in this project, and also it'll probably be better to achieve "`gnu+11` in NaCl, and `c++11` in WebAssembly". I'll look into this later.